### PR TITLE
Fix cosign errors with image digest

### PR DESCRIPTION
**Pull request by jimmykarily**
Part of https://github.com/kairos-io/kairos/issues/1005

### CODE REVIEW

File: .github/workflows/image.yaml [hunk 1]
```diff
@@ -96,10 +96,11 @@ jobs:
         if: ${{ github.event_name == 'push' && github.ref == 'refs/heads/main' }}
         env:
           COSIGN_YES: true
-          IMAGE: "quay.io/kairos/kairos-${{ matrix.flavor }}:latest"
+          IMAGE: "quay.io/kairos/kairos-${{ matrix.flavor }}"
+          TAG: "latest"
```
Comment (mauromorales) [inline .github/workflows/image.yaml:100]: not sure I understand why we use latest here, while on the arm ones we use the GITHUB_REF

**Comment by mudler** [comment]
somehow the PR tried to push the image:

```
Share your logs with an Earthly account (experimental)! Register for one at https://ci.earthly.dev./
The push refers to repository [quay.io/kairos/kairos-opensuse-leap-arm-rpi]
beca736e7a90: Preparing
462c37bec68e: Preparing
4d09be973b66: Preparing
2f72dc7478b6: Preparing
38cefcd0e787: Preparing
c52fb45986d6: Preparing
338943003afe: Preparing
af2dc33daadd: Preparing
392cc2b1a3e9: Preparing
b9c6e840f538: Preparing
5f70bf18a086: Preparing
a77631b19253: Preparing
1f5b1df46711: Preparing
b262b3efc2e7: Preparing
d3183c4c8e6f: Preparing
e49ae16a9f21: Preparing
4257e7a22778: Preparing
1c5fc4678c6e: Preparing
0a8331cc770a: Preparing
a5b824446154: Preparing
e49ae16a9f21: Waiting
4257e7a22778: Waiting
1c5fc4678c6e: Waiting
0a8331cc770a: Waiting
a5b824446154: Waiting
392cc2b1a3e9: Waiting
b9c6e840f538: Waiting
5f70bf18a086: Waiting
a77631b19253: Waiting
1f5b1df46711: Waiting
b262b3efc2e7: Waiting
d3183c4c8e6f: Waiting
c52fb45986d6: Waiting
338943003afe: Waiting
af2dc33daadd: Waiting
unauthorized: access to the requested resource is not authorized
Error: Process completed with exit code 1.
```

File: .github/workflows/image-arm.yaml [hunk 2]
```diff
@@ -38,6 +38,8 @@ jobs:
           platforms: all
       - name: Install Cosign
         uses: sigstore/cosign-installer@main
+      - name: Login to Quay Registry
```
Comment (mudler) [inline .github/workflows/image-arm.yaml:41]: we _don't_ have to push anything in the registry while building PRs

File: .github/workflows/image-arm.yaml [hunk 3]
```diff
@@ -47,7 +49,13 @@ jobs:
           MODEL: ${{ matrix.model }}
         run: |
           export TAG=${GITHUB_REF##*/}
-          ./earthly.sh +all-arm --IMAGE_NAME=kairos-$FLAVOR-$TAG.img --IMAGE=quay.io/kairos/kairos-$FLAVOR:$TAG --MODEL=$MODEL --FLAVOR=$FLAVOR
+          export IMAGE=quay.io/kairos/kairos-$FLAVOR
+
+          ./earthly.sh +all-arm --IMAGE_NAME=kairos-$FLAVOR-$TAG.img --IMAGE="$IMAGE:$TAG" --MODEL=$MODEL --FLAVOR=$FLAVOR
+
+          docker push "$IMAGE:$TAG" # Otherwise .RepoDigests will be empty for some reason
```
Comment (mudler) [inline .github/workflows/image-arm.yaml:56]: the problem here is that we always push. we should just push on master

Comment (jimmykarily) [reply .github/workflows/image-arm.yaml:56]: If we are not pushing, why are we signing?

File: .github/workflows/image-arm.yaml (hunk 2 above)
Comment (jimmykarily) [reply .github/workflows/image-arm.yaml:41]: added condition to push only on `main`.

File: .github/workflows/image-arm.yaml (hunk 3 above)
Comment (jimmykarily) [reply .github/workflows/image-arm.yaml:64]: added condition to sign and push only on `main`

File: .github/workflows/image.yaml (hunk 1 above)
Comment (jimmykarily) [reply .github/workflows/image.yaml]: changed to `GITHUB_REF` everywhere

**Review by mudler — APPROVED** [review]

**Pull request merged**